src/main/java/com/lob/net/ResponseGetter.java

```diff
@@ -0,0 +1,187 @@
+        return flattenParamsMap(params, null);
+    }
+
+    private static List<Parameter> flattenParamsMap(Map<String, Object> params, String keyPrefix) {
+        List<Parameter> flatParams = new LinkedList<Parameter>();
+        if (params == null) {
+            return flatParams;
+        }
+
+        for (Map.Entry<String, Object> entry : params.entrySet()) {
+            String key = entry.getKey();
+            Object value = entry.getValue();
+
+            String newPrefix = key;
+            if (keyPrefix != null) {
+                newPrefix = String.format("%s[%s]", keyPrefix, key);
+            }
+
+            flatParams.addAll(flattenParamsValue(value, newPrefix));
+        }
+
+        return flatParams;
+    }
+
+    private static List<Parameter> flattenParamsValue(Object value, String keyPrefix) {
+        List<Parameter> flatParams;
+
+        if (value instanceof Map<?, ?>) {
+            flatParams = flattenParamsMap((Map<String, Object>) value, keyPrefix);
+        } else {
+            flatParams = new LinkedList<Parameter>();
+            flatParams.add(new Parameter(keyPrefix, value.toString()));
+        }
+
+        return flatParams;
+    }
+
+    private static <T> LobResponse makeURLConnectionRequest(APIResource.RequestMethod method, Class<T> clazz, String url, String query, RequestOptions options) throws APIException, IOException {
+        java.net.HttpURLConnection conn = null;
+        try {
+            if (method == GET) {
+                conn = createGetConnection(url, query, options);
+            } else {
+                conn = createGetConnection(url, query, options);
+            }
+
+            int responseCode = conn.getResponseCode();
+
+            if (responseCode >= 200 && responseCode < 300) {
+                Map<String, List<String>> headers = conn.getHeaderFields();
+                T value = MAPPER.readValue(conn.getInputStream(), clazz);
+
+                return new LobResponse<T>(responseCode, value, headers);
+            } else {
+                throw MAPPER.readValue(conn.getErrorStream(), APIException.class);
+            }
+        } catch (IOException e) {
+            throw e;
+        } finally {
+            if (conn != null) {
+                conn.disconnect();
+            }
+        }
+    }
+
+    private static <T> LobResponse _request(APIResource.RequestMethod method,
+                                  String url, Map<String, Object> params, Class<T> clazz,
+                                  APIResource.RequestType type, RequestOptions options) throws AuthenticationException, APIException, IOException {
+        if (options == null) {
+            options = RequestOptions.getDefault();
+        }
+
+        String apiKey = options.getApiKey();
+        if (apiKey == null || apiKey.trim().length() == 0) {
+            throw new AuthenticationException("Missing API Key. (HINT: set your API key using 'Lob.init(<API_KEY>);')", null);
```

Comment:
nit: Not sure how I feel about about the word `HINT`. I'd expect that in a video game. Maybe something like:

`Missing API Key. Make sure 'Lob.init(<API_KEY>)' is called with a key from your Dashboard.`